### ISSUE TITLE
"Signals delivery fails constantly at GC #x" error on BSD

### ISSUE THREAD

**Issue by spaghettisalat** [ivmai/bdwgc#571]
With the lastest 8.2.4 release of bdwgc, some multithreading tests in ECL fail on FreeBSD, NetBSD and OpenBSD with an error `Signals delivery fails constantly at GC #x` (where `x` stands for some number) while older releases of bdwgc work fine.

I have tracked this down to commit 272e61e4691a9e990fa94a071d739dfbfa836941. The error disappears when compiling with the `-DGC_PTHREAD_SIGMASK_NEEDED` option (at least on FreeBSD and NetBSD) which supports the theory that this commit is to blame.

**Comment by ivmai** [comment]
Oh, it's a tricky thing :(
It looks like the fix should be in gc_pthread_redirects.h:
```
#   else
#     define GC_NO_PTHREAD_SIGMASK
```
->
```
#   elif !defined(GC_BUILD)
#     define GC_NO_PTHREAD_SIGMASK
```

**Comment by ivmai** [comment]
But let me check it in more detail

**Comment by ivmai** [comment]
In case of OpenBSD, there's an additional fix (commit 6f8349aba) to define GC_pthread_sigmask().

**Comment by spaghettisalat** [comment]
I can confirm that the fix works for FreeBSD and OpenBSD, but for NetBSD 9.3 I still get the same error.

**Comment by ivmai** [comment]
Hmm, do you have a workaround already?

**Comment by spaghettisalat** [comment]
On NetBSD 9.3, the situation is the same as before the fix: Everything works fine when I compile with `-DGC_PTHREAD_SIGMASK_NEEDED` but fails with the error "Signals delivery fails constantly at GC #x" otherwise.

**Comment by ivmai** [comment]
Please send me contents of pthreads.h, signal.h and sys/cdefs.h of NetBSD 9.3 /usr/include.

**Comment by ivmai** [comment]
> Please send me contents of pthreads.h, signal.h and sys/cdefs.h of NetBSD 9.3 /usr/include.

pthread_sigmask() prototype on NetBSD could be found here: https://github.com/IIJ-NetBSD/netbsd-src/blob/master/include/signal.h

**Comment by ivmai** [comment]
> On NetBSD 9.3, the situation is the same as before

Should be fixed by commit [1d98269](https://github.com/ivmai/bdwgc/commit/1d9826992aa2a2783c9badbacddae70fdf7fffee). Please test it.
